### ISSUE TITLE
Explicitly setting current page

### ISSUE THREAD

**Issue by marcelmika** [justinvallely/SwiftPhotoGallery#25]
Hi!

First, thanks a lot for a great lib!

Now, I have a question. Is there a way how to explicitly say which image should be shown when the user opens the gallery? e.g. I have a collection of several thumbnails and I want to open a gallery scrolled to the specific image. Is this possible? I was trying to setup the currentPage property before I present the gallery but it always shows the first image.

Thanks!

**Comment by justinvallely** [comment]
Yes, this is indeed possible. You're using the correct property so the problem probably is around how you tell it what page to go to. If you're using a UICollectionView for your thumbnails, just use the indexPath of the selected image to tell the SwiftPhotoGallery what page to go to:

```
    let gallery = SwiftPhotoGallery(delegate: self, dataSource: self)
    gallery.modalPresentationStyle = UIModalPresentationStyle.FullScreen

    presentViewController(gallery, animated: false, completion: { () -> Void in
        gallery.currentPage = indexPath.item
    })
```

**Comment by marcelmika** [comment]
Thanks a lot for a quick answer! 
This is exactly what I did, however, there is probably a small bug so I didn't know if this is a correct way how to do it. 

Open the demo app and try this: 

```
gallery.modalPresentationStyle = UIModalPresentationStyle.FullScreen        
presentViewController(gallery, animated: false, completion: { () -> Void in
     gallery.currentPage = 2
})
```

This works great but if you want to scroll to the previous image it goes directly to the first one. Basically, it skips one image if you are trying to move to the previous image. If you e.g. set the currentPage to 0 and move to next image everything works well.

It also doesn't happen when you scroll slowly.

**Comment by justinvallely** [comment]
I'm seeing this now too. This is a new bug that I believe to be a result of my recent modification to the page calculation method. It appears that it scrolls 2 images, so in the example app that only has 3 images, it would appear to always jump to the first. Thanks for bringing this to my attention.

**Comment by NikKovIos** [comment]
It still don't work properly and it is the reason of using another lib(

**Comment by justinvallely** [comment]
@NikKovV This is now fixed in v2.0.5
